### ISSUE TITLE
What happened to kairosdb.datastore.cassandra.write_delay

### ISSUE THREAD

**Issue by Loki-Afro** [kairosdb/kairosdb#429]
Hi,

what happened to `kairosdb.datastore.cassandra.write_delay`? I used this property to "flush" the kairosdb cache to write earlier to cassandra. Now that property is gone(?) 

How can I accomplish the same with kairosdb 1.2?

**Comment by brianhks** [comment]
Doh, I didn't expose that.  So here is how it works.  The queue processor checks the available data in the queue and if it is less than kairosdb.queue_processor.min_batch_size it waits 500ms to see if more comes in before writing to C*.  I can expose the delay but it only takes effect if you have a low volume of data coming int to Kairos.

**Comment by Loki-Afro** [comment]
are those 500ms configureable too?

Got some integration tests where I have to ensure some timing ;)

**Comment by brianhks** [comment]
No the 500ms is not configurable, but if you set the min_batch_size to 1 it will send whatevers is in the queue directly without delay.  I'll make the delay configurable in the 1.2.1 patch release.

**Comment by gmuehlenberg** [comment]
Just updated to 1.2.0. I have set the min_batch_size to 1, but still, my data is not being sent before the 500ms delay. Is anyone else experiencing this? Am I missing something?

**Comment by brianhks** [comment]
It is possible if you are only testing one single event that you still see a delay.  The queue processor is in a loop checking the queue for data points.  If the number of datapoints in the queue is less than min_batch_size it waits 500ms.  So if just 1ms before you send your single data point the thread checks the queue and sees 0 which is less than the min value it then waits 500ms and your data point comes in just after it goes into the wait.

I originally coded it to block on the queue until someone put something into it.  I saw strange data flow patterns as the batch sizes were always 1, 1000, 1, 5342, 1, 232, 1...

It kept triggering off the first point entered and sent it off as a lowly batch.  

Basically I'm saying the delay will go away as you have more data going through the system, because every time the processor thread comes back it will see data and never sleep.

That said I need to make a 1.2.1 release as someone found a bug in the string data so I will include a fix to make this value (500ms) configurable.

**Comment by gmuehlenberg** [comment]
Very well - thank you for the explanation! We'll just wait 500ms after test inserts till the next release. Have a nice weekend.
